### PR TITLE
Miscellaneous fixes for dqm online clients

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

The night of 13th March 2024, [run 377827](https://cmsoms.cern.ch/cms/runs/report?cms_run=377827) was mistakenly acquired using the DQM DAQ key `hi_run` (despite having other collision keys).
This uncovered a couple of leftovers from PRs https://github.com/cms-sw/cmssw/pull/43846 and https://github.com/cms-sw/cmssw/pull/43257 that made few DQM online clients crash (see logs at [DQM square](https://cmsweb.cern.ch/dqm/dqm-square/?run=377827&db=production)).
These are fixed here. 
 
#### PR validation:

```
python3 beam_dqm_sourceclient-live_cfg.py runkey=hi_run
python3 beampixel_dqm_sourceclient-live_cfg.py runkey=hi_run
python3 pixellumi_dqm_sourceclient-live_cfg.py runkey=hi_run
```

all compile fine now.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport, but will need to be backported to CMSSW_14_0_X for 2024 data-taking purposes.

### CODE REVIEW

**Comment by mmusich** [comment]
@fabiocos FYI

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by fabiocos** [comment]
the problem is due to the fact that with ```runkey=hi_run``` the logic building the ```fillDescription``` in https://github.com/cms-sw/cmssw/blob/c89c5e2fb17f13ed2614589562253cea8842fb29/RecoVertex/PrimaryVertexProducer/plugins/PrimaryVertexProducer.cc#L445 is selecting from the original ```offlinePrimaryVertices```

```
    TkClusParameters = cms.PSet(
        TkGapClusParameters = cms.PSet(
            zSeparation = cms.double(1.0)
        ),
        algorithm = cms.string('gap')
    ),
```

so clearly an attempt to update ```TkDAClusParameters``` does not work. So the question is whether the logic that select the algorithm ```gap``` for HI is holding or not. Apparently it is not, if we need this fix. @mandrenguyen any comment about this?

**Comment by mmusich** [comment]
> So the question is whether the logic that selects the algorithm gap for HI is holding or not.

If we are using the `gap` algo for offline reco, we certainly can use the same for DQM online (it actually makes more sense to me), but the beamspot people should chime in here as well (@francescobrivio FYI).

**Comment by mandrenguyen** [comment]
> > The full set of modifications to the vertexing for pp_on_AA may be found starting from this line (mainly for the benefit of others trying to follow this exchange):
> 
> indeed this is what I would naively do as well for the pixel vertices used in online DQM, but the beamspot people (or do we have a dedicated HI beamspot contact) need to confirm.

From HIN there is just an alca contact (@ashleyahram) but I don't think she'll know how the vertex reco for the beamspot workflow should be configured. 
Indeed if someone from the beamspot side can confirm that don't require settings different from the offline heavy-ion ones, I guess that would settle it.  At least from the reco/HI side, I think it should be set consistently.

File: RecoVertex/PrimaryVertexProducer/python/OfflinePixel3DPrimaryVertices_cfi.py [hunk 1]
```diff
@@ -12,12 +12,26 @@
...
+            zrange = cms.double(4.),          # consider only clusters within 4 sigma*sqrt(T) of a track
+            delta_highT = cms.double(1.e-2),  # convergence requirement at high T
+            delta_lowT = cms.double(1.e-3),   # convergence requirement at low T
+            convergence_mode = cms.int32(0),  # 0 = two steps, 1 = dynamic with sqrt(T)
+            Tmin = cms.double(2.4),           # end of vertex splitting, (adjusted for Pixel vertices)
+            Tpurge = cms.double(2.0),         # cleaning
+            Tstop = cms.double(0.5),          # end of annealing
+            vertexSize = cms.double(0.01),    # added in quadrature to track-z resolution (adjusted for Pixel vertices)
+            d0CutOff = cms.double(4.),        # downweight high IP tracks (adjusted for Pixel vertices)
+            dzCutOff = cms.double(3.),        # outlier rejection after freeze-out (T<Tmin)
```
Comment (mmusich) [inline RecoVertex/PrimaryVertexProducer/python/OfflinePixel3DPrimaryVertices_cfi.py:28]:
```suggestion
            dzCutOff = cms.double(4.),        # outlier rejection after freeze-out (T<Tmin) (adjusted for Pixel vertices)
```

**Review by mmusich — COMMENTED** [review]
swapped by mistake `d0` with `dz`

**Comment by mmusich** [comment]
>  Indeed if someone from the beamspot side can confirm that don't require settings different from the offline heavy-ion ones, I guess that would settle it. At least from the reco/HI side, I think it should be set consistently.

In case it helps for testing I prepared an alternative branch with the change of the clusterizer algo to gap for the HI case [here](https://github.com/cms-sw/cmssw/compare/master...mmusich:cmssw:mm_misc_fixes_forDQMonline_withGapForHI?expand=1).
This would change the status quo, so I would defer it to another later PR and go ahead with this change - which is purely technical to avoid further crashes at P5 (as fortuitous at they might be).

**Comment by mmusich** [comment]
type bug-fix

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mandrenguyen** [comment]
> > At least from the reco/HI side, I think it should be set consistently.
> 
> notice that these are pixel vertices, so I don't know how appropriate are in general the settings used in offline HI reco for full vertices.
> 
> _EDIT_: in this sense this PR maintains the status quo ante PR #43846

Thanks for pointing that out.  The following belongs to the old HI reco, which at some point was the only HI reco we had:
https://github.com/cms-sw/cmssw/blob/master/RecoHI/HiTracking/python/HIPixelAdaptiveVertex_cfi.py
Since the gap algo was used here, it must have also been run in beamspot workflows unless they were restricted to peripheral events, where pp reconstruction could be run. 
I think it's fairly safer to always use `gap` for heavy ions.

**Comment by mandrenguyen** [comment]
+reconstruction
restores heavy-ion like vertex reconstruction which was unintentionally removed

**Comment by tjavaid** [comment]
+1

**Comment by antoniovilela** [comment]
+1

**Pull request merged**